lib/ProductOpener/Orgs.pm

```diff
@@ -175,28 +175,30 @@ sub store_org ($org_ref) {
 
 		# We switched to validated, update CRM
 		my $main_contact_user = $org_ref->{main_contact};
-		my $user_ref = retrieve_user($main_contact_user);
 
 		eval {
-			my $contact_id = find_or_create_contact($user_ref);
-			defined $contact_id or die "Failed to get contact";
-			$user_ref->{crm_user_id} = $contact_id;
-			store_user($user_ref);
+			my $partner_id;
+			if (defined $main_contact_user) {
+				my $user_ref = retrieve_user($main_contact_user);
+				$partner_id = find_or_create_contact($user_ref);
```

Comment:
Why don't you do it only if $user_ref{crm_user_id} is undef ?
```suggestion
				$partner_id = $user_ref->{crm_user_id} // find_or_create_contact($user_ref);
```

Reply:
Sure we can, I just have to add a way to verify that it is a valid odoo record because this field can be changed manually by off admins in the user form. I'll do this in a new PR.